src/transformers/models/auto/modeling_auto.py

```diff
@@ -1207,6 +1208,12 @@
     ]
 )
 
+MODEL_FOR_KEYPOINT_DETECTION_MAPPING_NAMES = OrderedDict(
+    [
+        ("superpoint", "SuperPointModel"),
+    ]
+)
```

Comment:
I realise this was originally my suggestion, but reflecting and looking at this again I think we don't need to add this special class. We can remove `MODEL_FOR_KEYPOINT_DETECTION_MAPPING_NAMES` and just use `AutoModel` where needed

Reply:
I believe this is fixed with [this commit](https://github.com/huggingface/transformers/pull/28966/commits/4c3cc2e92f1edc5f0c5b7f4db679b328f56aac48)